pylint/checkers/base/basic_checker.py

```diff
@@ -444,6 +451,14 @@ def visit_expr(self, node: nodes.Expr) -> None:
             self.add_message("pointless-string-statement", node=node)
             return
 
+        # Warn W0133 for exceptions that are used as statements
+        if isinstance(expr, nodes.Call):
+            inferred = utils.safe_infer(expr)
```

Comment:
This `utils.safe_infer` invocation seems to retrieve the first item from a generator.

During the `string_formatting` functional test, [these inference tests](https://github.com/PyCQA/pylint/blob/c61bc76806f7b389db4b7b484174dbc286fa2c11/tests/functional/s/string/string_formatting.py#L47-L48) appear to fail as a result of that.  Instead of `0.missing` accessing the attribute with the name 'missing' from the [`ReturnYes` class](https://github.com/PyCQA/pylint/blob/c61bc76806f7b389db4b7b484174dbc286fa2c11/tests/functional/s/string/string_formatting.py#L29-L31), there is an to access the same attribute from the integer with value `2` (from the previous test fixture perhaps?)

The underlying exception is a result of a [`StopIteration` exception handled in `astroid` here](https://github.com/PyCQA/astroid/blob/8bdec591f228e7db6a0be66b6ca814227ff50001/astroid/decorators.py#L146-L148).

Reply:
Hm, I could be mistaken about the source of the `StopIteration`.  I forget exactly what brought me to that part of the code.

Reply:
Inference tips and/or caching are something to do with the problem here, I think.

* Removing the `str.format` inference tip from https://github.com/PyCQA/astroid/blob/8bdec591f228e7db6a0be66b6ca814227ff50001/astroid/brain/brain_builtin_inference.py#L1002-L1004 allows the `string_formatting` test to succeed -- but doing so causes other functional tests (such as `alternative_union_syntax`) begin to fail.

* Alternatively, adding `AttributeError` to the [list of exceptions handled in `astroid` here](https://github.com/PyCQA/astroid/blob/8bdec591f228e7db6a0be66b6ca814227ff50001/astroid/brain/brain_builtin_inference.py#L957) also works.

Either way: I don't understand why performing an additional inference on `Call` objects affects this.

Reply:
This issues has been resolved, right?

Reply:
Yep, this was resolved by PyCQA/astroid#1903.